frontend/src/pages/Metrics/index.tsx

```diff
@@ -1,110 +1,85 @@
 import Spinner from 'components/Spinner';
-import { SKIP_ONBOARDING } from 'constants/onboarding';
 import ResourceAttributesFilter from 'container/MetricsApplication/ResourceAttributesFilter';
-import MetricTable from 'container/MetricsTable';
+import MetricsTable from 'container/MetricsTable';
 import { convertRawQueriesToTraceSelectedTags } from 'lib/resourceAttributes';
-import React, { useEffect, useMemo } from 'react';
-import { connect, useSelector } from 'react-redux';
-import { bindActionCreators, Dispatch } from 'redux';
-import { ThunkDispatch } from 'redux-thunk';
-import { GetService, GetServiceProps } from 'store/actions/metrics';
+import React, { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
 import { AppState } from 'store/reducers';
-import AppActions from 'types/actions';
 import { GlobalReducer } from 'types/reducer/globalTime';
 import MetricReducer from 'types/reducer/metrics';
 import { Tags } from 'types/reducer/trace';
 
-function Metrics({ getService }: MetricsProps): JSX.Element {
-	const { minTime, maxTime, loading, selectedTime } = useSelector<
-		AppState,
-		GlobalReducer
-	>((state) => state.globalTime);
-	const {
-		services,
-		resourceAttributeQueries,
-		error,
-		errorMessage,
-	} = useSelector<AppState, MetricReducer>((state) => state.metrics);
+function Metrics(): JSX.Element {
+	const { minTime, maxTime } = useSelector<AppState, GlobalReducer>(
+		(state) => state.globalTime,
+	);
 
-	useEffect(() => {
-		if (error) {
-			notification.error({
-				message: errorMessage,
-			});
-		}
-	}, [error, errorMessage]);
+	const { resourceAttributeQueries } = useSelector<AppState, MetricReducer>(
+		(state) => state.metrics,
+	);
 
 	const selectedTags = useMemo(
 		() =>
 			(convertRawQueriesToTraceSelectedTags(resourceAttributeQueries) as Tags[]) ||
 			[],
 		[resourceAttributeQueries],
 	);
-	const isSkipped = getLocalStorageKey(SKIP_ONBOARDING) === 'true';
-
-	useEffect(() => {
-		if (loading === false) {
-			getService({
-				maxTime,
-				minTime,
-				selectedTags,
-			});
-		}
-	}, [getService, loading, maxTime, minTime, selectedTags]);
 
-	useEffect(() => {
-		let timeInterval: NodeJS.Timeout;
+	const { t } = useTranslation('common');
 
-		if (loading === false && !isSkipped && services.length === 0) {
-			timeInterval = setInterval(() => {
+	const { data, isLoading, isError, refetch, status } = useQuery(
+		['services', [maxTime, minTime, selectedTags]],
+		{
+			queryFn: () =>
 				getService({
-					maxTime,
-					minTime,
+					end: maxTime,
+					start: minTime,
 					selectedTags,
-				});
-			}, 50000);
-		}
+				}),
+		},
+	);
 
-		return (): void => {
-			clearInterval(timeInterval);
-		};
-	}, [
-		getService,
-		isSkipped,
-		loading,
-		maxTime,
-		minTime,
-		services,
-		selectedTime,
-		selectedTags,
-	]);
+	// api failed to load the data
+	if (isError) {
```

Comment:
no need for this check we can show the table with whatever data it has but no need to hide it please show toast in case of error